src/main/java/com/dataloom/neuron/Neuron.java

```diff
@@ -51,14 +78,65 @@ public void activateReceptor( SignalType type, Receptor receptor ) {
         }
     }
 
-    public void transmit( AuditableSignal signal ) {
+    public void transmit( Signal signal ) {
+
+        // 1. write to the Audit Entity Set
+        UUID auditId = writeToAuditEntitySet( signal );
+
+        if ( auditId != null ) {
 
-        // 1. audit event
-        this.auditLogQueryService.store( signal );
+            // 2. write to the Audit Log Table
+            writeToAuditLog( signal, auditId );
+        }
+
+        // 3. hand off event to receptors
+        Set<Receptor> receptors = this.receptors.get( signal.getType() );
+
+        // TODO: does order matter?
+        // TODO: what about parallelization?
+        receptors.forEach( receptor -> receptor.process( signal ) );
+    }
 
-        // 2. hand off event to receptors
-        // List<Receptor> receptors = this.receptors.get( signal.getType() );
-        // receptors.forEach( synapse -> synapse.process( signal ) );
+    private UUID writeToAuditEntitySet( Signal signal ) {
+
+        try {
+
+            UUID auditEntitySetId = AuditEntitySet.getId();
+            UUID auditEntitySetSyncId = dataSourceManager.getCurrentSyncId( auditEntitySetId );
```

Comment:
I agree that it makes sense to have the `syncId` be inside `AuditEntitySet`. However, it would have to be static, in which case, how would we do `getCurrentSyncId()`?